pkg/addons/addons_gcpauth.go

```diff
@@ -80,9 +80,9 @@ func enableAddonGCPAuth(cfg *config.ClusterConfig) error {
 					creds, _ = google.CredentialsFromJSON(ctx, f)
 				}
 			}
-		} else {
-			exit.Message(reason.InternalCredsNotFound, "Could not find any GCP credentials. Either run `gcloud auth application-default login` or set the GOOGLE_APPLICATION_CREDENTIALS environment variable to the path of your credentials file.")
 		}
+	} else {
+		exit.Message(reason.InternalCredsNotFound, "Could not find any GCP credentials. Either run `gcloud auth application-default login` or set the GOOGLE_APPLICATION_CREDENTIALS environment variable to the path of your credentials file.")
```

Comment:
On second look, this doesn't seem to be doing the right thing. This `else` will get called if err is nil and the creds json is not nil, which is exactly the opposite of what we want. 

We should be getting by this if statement if your service account is authenticated correctly and the JSON is in a place we're expecting.

Reply:
You're right, I rushed a little bit to unblock us.

Let me push a new commit.

Reply:
@sharifelgamal let me know what do you think now, basically if we have an error when retrieving credentials or they are empty, we first check if it is running in CloudShell or GCE, if its not the case we fail with `InternalCredsNotFound`.